verbit/streaming_client.py

```diff
@@ -342,3 +371,41 @@ def _get_ws_connect_query_string(media_config: MediaConfig, response_types: Resp
 
     def _get_ws_auth_info(self) -> dict:
         return {'Authorization': f'Bearer {self._ws_access_token}'}
+
+class WebSocketStreamingClient_Reconnect(WebSocketStreamingClient_Vanilla):
+    """Wrap the Vanilla client to reconnect to server and continue.
+
+    Notes:
+     1. Handles the case where the server unexpectedly closed connection (for any reason)
+     2. Temporary lack of connection is often handled in the TCP layer,
+        that is, disconnecting and reconnecting from a network does not mean issue a disconnect, and is still handles in the Vanilla case as well.
+     3. Changing a logical network, such as the client being assigned a new IP, connecting to a different network
+        is not handled in this case, since the connection to the server does not fail yet, even though another route
+        might already be available. That case needs some Watch-Dog style task.
+    """
+
+    def __init__(self, access_token, on_media_error: typing.Callable[[Exception], None] = None):
+        super().__init__(access_token, on_media_error)
+
+    def start_stream(self,
+                     media_generator,
+                     media_config: MediaConfig = None,
+                     response_types: ResponseType = ResponseType.Transcript) -> typing.Generator:
+
+        response_generator = super().start_stream(media_generator, media_config, response_types)
+
+        def reconnect_generator():
+            try:
+                self._logger.debug(f'Starting reconnect_generator: with WS instance: {id(self._ws_client)}')
+                yield from response_generator
+
+            except (ConnectionError, WebSocketException) as connection_error:
+                self._logger.debug(f'caught {type(connection_error)} {connection_error=}, will try reconnect-and-continue')
+                yield from self.start_stream(media_generator, media_config, response_types)
+
+        return reconnect_generator()
+
```

Comment:
Intuitively, I firstly returned the closure above (`reconnect_generator()`), then I removed it, my local test (streaming, killing local server and re-running it) : Works!
But: Unit-tests fail... , I don't understand the nuance now...